lib/jekyll/utils.rb

```diff
@@ -297,5 +287,36 @@ def merged_file_read_opts(site, opts)
       merged
     end
 
+    private
+    def merge_values(target, overwrite)
+      target.merge!(overwrite) do |_key, old_val, new_val|
+        if new_val.nil?
+          old_val
+        elsif mergable?(old_val) && mergable?(new_val)
+          deep_merge_hashes(old_val, new_val)
+        else
+          new_val
+        end
+      end
+    end
+
+    private
+    def merge_default_proc(target, overwrite)
+      if default_proc?(target) && default_proc?(overwrite) && target.default_proc.nil?
+        target.default_proc = overwrite.default_proc
+      end
+    end
+
+    private
+    def default_proc?(object)
```

Comment:
The original code was:
```ruby
if target.respond_to?(:default_proc) && overwrite.respond_to?(:default_proc) && target.default_proc.nil?
  target.default_proc = overwrite.default_proc
end
```
I couldn't find the way to make `if` statement line shorter, so I created this `default_proc?` method.
Do you suggest to change the `if` condition to check if both `target` and `overwrite` are `Hash`?
Like this:
```ruby
if target.is_a?(Hash) && overwrite.is_a?(Hash) && target.default_proc.nil?
  target.default_proc = overwrite.default_proc
end
```